web/hooks/push.go

```diff
@@ -0,0 +1,286 @@
+				results := wh.runAssignmentTests(assignment, repo, course, payload)
+				if !isDefaultBranch(payload) {
+					// Attempt to find the pull request for the branch, if it exists,
+					// and then assign reviewers to it, if the branch task score is higher than the assignment score limit
+					wh.handlePullRequestPush(payload, results, assignment, course, repo)
+				}
+			} else {
+				wh.logger.Debugf("Ignoring assignment: %s, pushed to group repo: %s", assignment.GetName(), payload.GetRepo().GetName())
+			}
+		}
+	default:
+		wh.logger.Debug("Nothing to do for this push event")
+	}
+}
+
+// handlePullRequestPush attempts to find a pull requested associated with the non-default branch.
+// If successfull, it then finds the relevant task, and uses it to receive the relevant task score.
+// If a passing score is reached, it assigns reviewers to the pull request.
+// It also uses the test results and task to generate a feedback comment for the pull request.
+func (wh GitHubWebHook) handlePullRequestPush(payload *github.PushEvent, results *score.Results, assignment *pb.Assignment, course *pb.Course, repo *pb.Repository) {
+	wh.logger.Debugf("Attempting to find pull request for ref: %s, in repository: %s",
+		payload.GetRef(), payload.GetRepo().GetFullName())
+
+	pullRequest, err := wh.db.GetPullRequest(&pb.PullRequest{
+		SourceBranch:         branchName(payload.GetRef()),
+		ExternalRepositoryID: uint64(payload.GetRepo().GetID()),
+	})
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			// This can happen if someone pushes to a branch group assignment, without having a PR created for it
+			// If this happens, QF should not do anything
+			wh.logger.Debugf("No pull request found for ref: %s", payload.GetRef())
+			return
+		}
+		wh.logger.Errorf("Failed to get pull request from database: %v", err)
+		return
+	}
+	tasks, err := wh.db.GetTasks(&pb.Task{ID: pullRequest.GetTaskID()})
+	if err != nil {
+		// A pull request should always have a task association
+		// If not, something must have gone wrong elsewhere
+		wh.logger.Errorf("Failed to get task from the database: %v", err)
+		return
+	}
+	if len(tasks) != 1 {
+		// This should never happen
+		wh.logger.Errorf("Got an unexpected number of tasks: %d", len(tasks))
+		return
+	}
+	task := tasks[0]
+	// TODO(meling): My idea is that when a teacher wants to assign a test to a specific task they will use the 'local' task name.
+	// For example, if a teacher has created the markdown file task-hello_world.md,
+	// they would do scores.AddWithTaskName(TestHelloWorld, "hello_world", max, weight).
+	// Do you concur with this approach?
+	// We could of course simply make teachers assign the global task name, but that seems somewhat counter-intuitive to me.
+
+	// TODO(espeland): Revise this when score task name format has been decided.
+	taskSum := results.TaskSum(task.LocalName())
+
+	// TODO(espeland): When the project is finished. Create a GitHub issue that states all places where
```

Comment:
Yes. You could mark it as a special TODO, e.g., `// TODO(Espeland) Update here for GitHub app.` and you could add the necessary context in a separate issue on the issue tracker.